analyzer-java/src/main/java/com/infosupport/ldoc/analyzerj/ldjexceptions/LdocException.java

```diff
@@ -0,0 +1,15 @@
+package com.infosupport.ldoc.analyzerj.ldjexceptions;
+
+/**
+ * General Exception that can be thrown by the Living Documentation Analyzer.
```

Comment:
I don't really follow this one - if it's really a general exception, what does this do that a RuntimeException does not?

When would code that calls the analyzer catch and handle `LdocException`s from the analyzer one way, and handle `RuntimeException`s from the analyzer another way? If that place does not exist, does the difference matter?

Is the intention that the analyzer will, in the future, only throw `LdocException` and nothing else? Or is this really a `MethodThatTakesNodeWasPassedANodeButItsTheWrongKindOfNodeException`, and intended to stay that way?

All in all, for me, this adds more confusion than it clears up; prefer remove.

Reply:
Throwing generic runtime exceptions is, I think, never a good practice. Also, as an exception should be specific, I agree that the current form of this custom exception is not good.

To mitigate this issue properly, I see two options:
1. Change the custom exception to something more specific, e.g., `MethodThatTakesNodeWasPassedANodeButItsTheWrongKindOfNodeException` but preferably a bit more concise
2. Throw an UnsupportedOperation exception with a message stating something like 'not yet implemented' (see https://stackoverflow.com/a/22045192 for a discussion) 

Curious to know how you think about this

Reply:
My preference would be to reshape the `resolve` code to use overloading, as I suggested over there. That way:

* the "it was passed the wrong subtype" case is guaranteed by the compiler not to happen
* no exception needs to be thrown in `resolve` at all
* this class can be removed entirely
* the decision whether it is better to throw an `RuntimeException` or a `MethodThatTakesNodeWasPassedANodeButItsTheWrongKindOfNodeException` can be left to some other time.